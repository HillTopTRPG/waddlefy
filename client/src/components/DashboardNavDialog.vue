<template>
  <v-dialog
    :model-value="modalValue"
    attach="#dashboard-nav"
    :fullscreen="true"
    :scrim="false"
    :no-click-animation="true"
    :persistent="true"
    class="mb-12 mt-12"
    transition="slide-y-reverse-transition"
  >
    <v-card>
      <v-toolbar color="secondary" class="px-2" density="compact" :title="title">
        <v-toolbar-items>
          <v-btn variant="text" icon="mdi-close" @click="emits('update:modalValue', !modalValue)" />
        </v-toolbar-items>
      </v-toolbar>
      <slot />
    </v-card>
  </v-dialog>
</template>

<script lang="ts" setup>
const props = defineProps<{
  modalValue: boolean
  title: string
}>()

const emits = defineEmits<{
  (e: 'update:modalValue', value: boolean): void
}>()
</script>
