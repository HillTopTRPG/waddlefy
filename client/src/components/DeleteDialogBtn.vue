<template>
  <v-dialog width="auto" v-model="deleteDialog">
    <template v-slot:activator="{ props }">
      <v-btn color="error" variant="tonal" v-bind="props">この{{ type }}を削除</v-btn>
    </template>
    <v-card>
      <v-card-title class="bg-warning">{{ targetName }} の削除</v-card-title>
      <v-card-text>
        <p>削除するにはこの{{ type }}の名前を入力してください。</p>
      </v-card-text>
      <v-card-item>
        <v-text-field v-model="inputTargetName" :placeholder="targetName"/>
      </v-card-item>
      <v-card-actions>
        <v-btn
          class="flex-grow-1"
          color="warning"
          variant="elevated"
          :disabled="inputTargetName !== targetName"
          text="完全に削除"
          @click="deleteExecute()"
        />
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script setup lang="ts">
import { ref, watch } from 'vue'
import 'splitpanes/dist/splitpanes.css'

const props = defineProps<{
  type: string
  targetName: string
  sessionId: string
}>()

const emits = defineEmits<{
  (e: 'execute'): Promise<void>
}>()

watch(() => props.sessionId, () => {
  deleteDialog.value = false
  inputTargetName.value = ''
})

const deleteDialog = ref(false)
const inputTargetName = ref('')
async function deleteExecute() {
  await emits('execute')
  deleteDialog.value = false
  inputTargetName.value = ''
}
</script>
