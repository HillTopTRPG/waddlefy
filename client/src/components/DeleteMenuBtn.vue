<template>
  <v-menu width="auto" v-model="opened" :close-on-content-click="false">
    <template #activator="{ props }">
      <v-btn data-cy="open-btn" :color="color || 'error'" variant="text" :class="classText || ''" v-bind="props">
        <v-icon icon="mdi-delete-outline" />
        <span data-cy="text" class="text-decoration-underline">{{
          $t('label.delete-this').replace('$$', i18n ? $t(type) : type)
        }}</span>
      </v-btn>
    </template>
    <v-card>
      <v-card-text class="pb-1">{{ $t('label.delete-for-message').replace('$$', i18n ? $t(type) : type) }}</v-card-text>
      <v-card-item class="pa-2">
        <v-text-field
          :label="$t('label.name-of').replace('$$', i18n ? $t(type) : type)"
          v-model="inputTargetName"
          :autofocus="true"
          variant="solo-filled"
          color="blue-gray-lighten-4"
          :flat="true"
          :hide-details="true"
          :persistent-placeholder="true"
          :placeholder="targetName"
          ref="inputElm"
        />
      </v-card-item>
      <v-divider />
      <v-card-actions class="px-2">
        <v-btn
          class="flex-0-1-100 text-decoration-underline"
          variant="text"
          @click="opened = false"
          :text="$t('label.cancel')"
        />
        <v-btn
          color="warning"
          class="flex-0-1-100"
          variant="flat"
          :text="$t('label.delete-completely')"
          :disabled="inputTargetName !== targetName"
          @click="deleteExecute()"
        />
      </v-card-actions>
    </v-card>
  </v-menu>
</template>

<script setup lang="ts">
import 'splitpanes/dist/splitpanes.css'
import { ref, watch } from 'vue'

defineProps<{
  type: string
  targetName: string
  classText?: string
  color?: string
  i18n?: boolean
}>()

const emits = defineEmits<{
  (e: 'execute'): Promise<void>
}>()

const opened = ref(false)
const inputTargetName = ref('')
const inputElm = ref()

async function deleteExecute() {
  opened.value = false
  await emits('execute')
  inputTargetName.value = ''
}

watch(opened, v => {
  if (v) {
    inputTargetName.value = ''
    setTimeout(() => inputElm.value?.focus(), 300)
  }
})
</script>
