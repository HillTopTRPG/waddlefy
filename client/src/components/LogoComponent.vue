<template>
  <v-sheet class="text-center">
    <svg
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
      version="1.1"
      :height="height"
      viewBox="0 0 640 640"
      xml:space="preserve"
      class="align-center"
      :style="`--img-color: ${color};`"
    >
      <desc>Created with Fabric.js 5.3.0</desc>
      <defs></defs>
      <g transform="matrix(1.1482580233 0 0 1.1482580233 225.6335008444 320.4962267981)" id="CiQZmcKFQacBrHscwjfs3"  >
        <path style="stroke: var(--img-color); stroke-width: 1; stroke-dasharray: none; stroke-linecap: butt; stroke-dashoffset: 0; stroke-linejoin: miter; stroke-miterlimit: 4; fill: var(--img-color); fill-rule: nonzero; opacity: 1;" vector-effect="non-scaling-stroke"  transform=" translate(-0.0000060855, -0.0000017674)" d="M 9.61664 -244.60607 C 6.68586 -230.58515999999997 18.81434 -212.31418 39.64154 -196.39987 C 41.35813 -195.0882 41.12645 -195.48257999999998 44.33946 -200.31574 C 47.55246 -205.1489 52.128080000000004 -204.65329 53.46194 -203.15398000000002 C 53.39358 -192.09883000000002 34.02173 -152.44581000000002 34.42579 -144.79828000000003 C 34.82986 -100.91222000000003 36.16369 -96.24135000000004 29.429949999999998 -90.42671000000004 C 76.92358999999999 106.11231999999995 12.320239999999998 140.89997999999997 4.956819999999997 134.75033999999997 C -2.4066100000000032 128.60069999999996 4.498789999999997 95.30825999999996 -0.4198400000000033 59.27896999999997 C -3.3308200000000032 43.64156999999997 -18.399700000000003 31.841349999999974 -18.258390000000006 23.320189999999975 C -18.117070000000005 14.799019999999976 -30.919130000000006 -23.819570000000027 -31.899450000000005 -23.409360000000028 C -32.879780000000004 -22.99915000000003 -22.266550000000006 17.48859999999997 -21.983860000000007 24.78656999999997 C -21.70117000000001 32.08453999999997 -12.084790000000007 43.46829999999997 -3.1223200000000055 60.33622999999997 C -0.2027100000000055 75.52803999999998 0.05708999999999431 101.33690999999997 -0.4198300000000055 118.52399999999997 C -71.96042 143.46820999999997 9.155239999999996 207.00577999999996 36.47541999999999 255.14305999999996 C 38.23843999999999 256.94282999999996 43.971959999999996 263.42950999999994 48.64604999999999 265.16565999999995 C 36.31470999999999 262.77324999999996 36.496619999999986 256.13089999999994 29.24913999999999 257.46595999999994 C 18.89759999999999 261.08226999999994 -92.74241 256.7348099999999 -105.08749000000003 276.84601999999995 C -162.30250000000004 273.52509 -169.84004000000004 281.13127999999995 -195.57889000000003 276.84601999999995 C -194.34805000000003 274.98500999999993 -186.88952000000003 266.99580999999995 -125.61866000000003 236.67468999999994 C -115.77918000000004 231.24058999999994 -112.28209000000004 226.75129999999993 -111.94253000000003 214.30055999999993 C -112.48798000000004 201.32209999999992 -110.01292000000004 150.93215999999993 -111.94253000000003 134.75033999999994 C -110.48014000000003 0.4961599999999464 -111.46883000000004 -67.96066000000008 -56.30792000000003 -181.9775400000001 C -42.04436000000003 -206.0791000000001 -18.65664000000003 -227.8951800000001 1.3204499999999655 -244.6060700000001 C 24.135259999999967 -263.0481000000001 14.186019999999965 -269.0135200000001 52.702679999999965 -275.2500900000001 C 77.64960999999997 -286.0668500000001 123.66260999999997 -263.7920400000001 130.80186999999995 -261.0406000000001 C 137.94111999999996 -258.2891600000001 149.05471999999995 -243.4782300000001 172.03731999999997 -208.5752800000001 C 185.81987999999996 -197.90545000000012 199.94371999999996 -177.69943000000012 194.30734999999996 -178.4267100000001 C 191.67736999999997 -178.7660600000001 178.93543999999997 -186.1406900000001 166.36401999999995 -189.7984200000001 C 188.63345999999996 -183.0655200000001 159.87539999999996 -210.0178400000001 129.50760999999994 -208.5752900000001 C 160.10606999999993 -201.5815500000001 180.73511999999994 -186.37995000000012 172.03731999999994 -188.0387100000001 C 158.07966999999994 -191.3196800000001 118.37410999999994 -192.4411900000001 98.64673999999994 -178.4267200000001 C 94.10128999999993 -182.6085400000001 72.67296999999994 -189.5796900000001 65.79358999999994 -203.1539900000001 C 60.83728999999994 -207.4975300000001 65.79358999999994 -211.8812600000001 65.79358999999994 -211.8812600000001 C 65.79358999999994 -211.8812600000001 79.94204999999994 -237.23409000000012 76.79913999999994 -246.8064900000001 C 73.65622999999994 -256.3788900000001 41.210889999999935 -270.9204900000001 25.172299999999936 -264.2231800000001 C 21.370519999999935 -260.72479000000016 12.938849999999936 -260.4995900000001 9.616649999999936 -244.60607000000013 z" stroke-linecap="round" />
      </g>
      <g transform="matrix(1.1482580233 0 0 1.1482580233 212.2788512766 375.1453937109)" id="n7jIBGEcDISmIEB7RGzTV"  >
        <path style="stroke: var(--img-color); stroke-width: 3; stroke-dasharray: none; stroke-linecap: butt; stroke-dashoffset: 0; stroke-linejoin: miter; stroke-miterlimit: 4; fill: var(--img-color); fill-opacity: 0; fill-rule: nonzero; opacity: 1;" vector-effect="non-scaling-stroke"  transform=" translate(-0.0000034519, 0.0000054291)" d="M 110.52402 -228.24579 C 113.20858 -221.41202 109.00296999999999 -191.68544 110.52402 -184.68178 C 128.82359 -11.23517000000001 77.91837999999998 137.92095 76.8021 141.01643 C 74.60358 147.79157 100.21521 190.51786 95.02958 196.95770000000002 C 93.95651 199.06854 74.43484 222.09397 65.30440999999999 223.00646 C 47.84064999999999 211.14118 39.948229999999995 199.64415 30.48205999999999 210.73013 C -4.539670000000008 224.03833 -93.20881000000001 230.94844 -114.48978 227.25945000000002" stroke-linecap="round" />
      </g>
      <g transform="matrix(1.008627294 0 0 1.008627294 341.8439650733 620.1282528782)" id="UcmS41-ig2BKsljRWAEfN"  >
        <path style="stroke: var(--img-color); stroke-width: 1; stroke-dasharray: none; stroke-linecap: butt; stroke-dashoffset: 0; stroke-linejoin: miter; stroke-miterlimit: 4; fill: var(--img-color); fill-rule: nonzero; opacity: 1;" vector-effect="non-scaling-stroke"  transform=" translate(0.0000124028, -0.0000030794)" d="M -23.89294 -15.65564 C -23.89294 -15.65564 -20.55961 -8.20772 -17.43461 -8.15564 C -14.30961 -8.10356 14.28414 -19.50981 25.273719999999997 -17.94731 C 39.596639999999994 -13.259810000000002 51.30865 0.6063399999999994 49.85705 1.8443599999999982 C 48.80386 2.883169999999998 33.86747 -3.520220000000002 25.27372 -0.2389700000000019 C 25.0133 1.167279999999998 31.546020000000002 5.619519999999998 30.48205 6.219359999999998 C 12.357050000000001 -3.5723100000000017 -18.89295 18.094359999999998 -31.39295 18.094359999999998 C -37.01795 18.094359999999998 -49.97896 8.452059999999998 -49.97896 8.452059999999998 C -49.97896 8.452059999999998 -43.26795 7.261019999999998 -23.89295 -15.655640000000004 z" stroke-linecap="round" />
      </g>
      <g transform="matrix(0.9754985665 0 0 1.0149660826 534.8707045488 443.6207751179)" id="cXz2YTpbbQMm-f1G5vmMM"  >
        <path style="stroke: var(--img-color); stroke-width: 1; stroke-dasharray: none; stroke-linecap: butt; stroke-dashoffset: 0; stroke-linejoin: miter; stroke-miterlimit: 4; fill: var(--img-color); fill-rule: nonzero; opacity: 1;" vector-effect="non-scaling-stroke"  transform=" translate(0.0000108117, -0.0000015441)" d="M -79.46648 -149.86487 C -77.69514000000001 -149.88457 -74.05495 -171.07524999999998 -72.51965 -170.81061 C -71.09546 -170.56512 -59.856849999999994 -157.54318 -58.492869999999996 -157.01077 C -55.097379999999994 -155.68539 -51.67538999999999 -153.40796 -47.166 -149.86487 C -39.3535 -165.22944999999999 -26.880429999999997 -170.81061 -26.880429999999997 -170.81061 C -26.880429999999997 -170.81061 -16.724179999999997 -156.15579 -26.880429999999997 -135.11122 C -23.764619999999997 -120.82398 -10.777299999999997 -130.70426 -12.429139999999997 -78.61985 C -12.429139999999997 -78.61985 11.427810000000004 -72.06897 15.051000000000002 -29.54139 C 123.72832 31.244639999999997 120.50086 139.04927 47.00229 169.36777999999998 C 7.723020000000005 178.28383999999997 -12.539760000000001 143.55433999999997 -7.469809999999995 130.61487999999997 C -9.646969999999996 120.93851999999997 -50.07456 72.65934999999996 -51.679159999999996 117.69520999999997 C -60.53332999999999 116.93541999999998 -99.14292 64.70782999999997 -99.76762 30.48900999999998 C -100.11361 11.53682999999998 -59.08048999999999 -43.135510000000025 -58.49287999999999 -67.84289000000003 C -57.905269999999994 -92.55028000000003 -25.495589999999993 -108.50967000000003 -26.880439999999993 -111.14485000000002 C -32.333749999999995 -121.52180000000001 -76.48642999999998 -103.93888000000003 -86.27606999999999 -121.30902000000002 C -96.06571 -146.38237 -81.88292999999999 -149.83798000000002 -79.4665 -149.86486000000002 z" stroke-linecap="round" />
      </g>
      <g transform="matrix(1.1423455128 0 0 1.1423455128 619.4879248642 472.3770964521)" id="rp5xkkkz82nwxHCWyU5_a"  >
        <path style="stroke: var(--img-color); stroke-width: 15; stroke-dasharray: none; stroke-linecap: butt; stroke-dashoffset: 0; stroke-linejoin: miter; stroke-miterlimit: 4; fill: var(--img-color); fill-opacity: 0; fill-rule: nonzero; opacity: 1;" vector-effect="non-scaling-stroke"  transform=" translate(-0.0000037967, 0)" d="M -10.93219 94.21673 C 13.748879999999998 77.91209 17.6441 -15.133380000000002 0.19595999999999947 -61.41538000000001 C -3.4060200000000007 -68.85123000000002 7.5863499999999995 -94.21673000000001 7.5863499999999995 -94.21673000000001" stroke-linecap="round" />
      </g>
      <g transform="matrix(0.2675743008 0 0 0.2675743008 627.5675991 364.9832308814)" id="A5h2P2k8I1MIfCQ94Grnp"  >
        <path style="stroke: var(--img-color); stroke-width: 0; stroke-dasharray: none; stroke-linecap: butt; stroke-dashoffset: 0; stroke-linejoin: miter; stroke-miterlimit: 4; fill: var(--img-color); fill-rule: nonzero; opacity: 1;" vector-effect="non-scaling-stroke"  transform=" translate(0, 0)" d="M 0 -32.89474 C 18.15789 -32.89474 32.89474 -18.157899999999998 32.89474 0 C 32.89474 18.15789 18.157899999999998 32.89474 0 32.89474 C -18.15789 32.89474 -32.89474 18.157899999999998 -32.89474 0 C -32.89474 -18.15789 -18.157899999999998 -32.89474 0 -32.89474 z" stroke-linecap="round" />
      </g>
      <g transform="matrix(1.1423455128 0 0 1.1423455128 499.9790503469 467.7669732879)" id="APnEXzCzGAG1KVJk_ciCl"  >
        <path style="stroke: var(--img-color); stroke-width: 3; stroke-dasharray: none; stroke-linecap: butt; stroke-dashoffset: 0; stroke-linejoin: miter; stroke-miterlimit: 4; fill: var(--img-color); fill-opacity: 0; fill-rule: nonzero; opacity: 1;" vector-effect="non-scaling-stroke"  transform=" translate(-0.0000057817, 0.0000027778)" d="M -44.295 -148.99984 C -44.295 -148.99984 -67.82419 -146.49404 -70.52689000000001 -133.86589 C -71.60283000000001 -128.83863 -79.35751 -122.57217 -73.15148 -114.96548000000001 C -63.768370000000004 -103.46465 -44.19332000000001 -87.55861000000002 -44.294990000000006 -82.27105000000002 C -44.463910000000006 -73.48627000000002 -71.32202000000001 -52.42404000000002 -76.05175 -11.504880000000014 C -77.4031 70.63048999999998 -51.10114 75.47685999999999 -44.29499 103.02718999999999 C -61.13749 134.63156999999998 -61.15824 124.8852 -73.15147999999999 131.12884 C -85.14471999999999 137.37248 -76.05175 146.92067 -76.05175 146.92067 C -76.05175 146.92067 -51.831379999999996 150.14084 -44.29499 146.93413 C -21.37399 129.78512 -13.213909999999998 82.17077 -13.213909999999998 82.17077 C -13.213909999999998 82.17077 1.559490000000002 86.67215 -8.484179999999999 127.75046 C -24.641389999999998 133.45758 -13.213909999999998 146.92067 -13.213909999999998 146.92067 C -13.213909999999998 146.92067 60.346959999999996 151.59879 64.48879 146.92067 C 68.63062 142.24255 63.39265999999999 140.2342 64.48879 136.94181 C 65.58492 133.64942 79.35364999999999 125.01748 79.35364999999999 125.01748" stroke-linecap="round" />
      </g>
    </svg>
  </v-sheet>
</template>

<script lang="ts" setup>
defineProps<{
  color: string,
  height: string
}>()
</script>
