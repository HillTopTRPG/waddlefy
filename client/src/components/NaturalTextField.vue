<template>
  <v-text-field
    :readonly="!editing"
    label=""
    v-model="inputValue"
    :placeholder="label"
    variant="plain"
    density="comfortable"
    class="name-text-field"
    style="letter-spacing: 1em; min-height: 1em"
    :hide-details="true"
    ref="textFieldElm"
    @keydown.enter="$event.keyCode === 13 && submit()"
    :autofocus="true"
    @click:control.stop
    @click:append-inner.stop
  >
    <template #append-inner>
      <template v-if="editable">
        <v-divider :vertical="true" />
        <v-defaults-provider :defaults="{ VBtn: { stacked: true, variant: 'text', size: 'x-small' } }">
          <v-btn
            v-if="editing"
            :disabled="!inputValue"
            prepend-icon="mdi-check-bold"
            text="決定"
            class="bg-transparent h-100 px-1"
            @click.prevent.stop="submit()"
            @mousedown.prevent.stop
            @mouseup.prevent.stop
          />
          <v-btn
            v-else
            prepend-icon="mdi-pencil"
            text="編集"
            class="bg-transparent h-100 px-1"
            @click.prevent.stop="editStart()"
            @mousedown.prevent.stop
            @mouseup.prevent.stop
          />
        </v-defaults-provider>
      </template>
    </template>
  </v-text-field>
</template>

<script lang="ts" setup>
import { ref, watch } from 'vue'

const props = defineProps<{
  value: string
  label: string
  editable: boolean
}>()

watch(
  () => props.value,
  v => {
    inputValue.value = v
  }
)

const emits = defineEmits<{
  (e: 'submit', value: string): Promise<void>
}>()

const editing = ref(false)
const textFieldElm = ref()
const inputValue = ref(props.value)
async function submit() {
  await emits('submit', inputValue.value)
  editing.value = false
}
function editStart() {
  editing.value = true
  setTimeout(() => {
    textFieldElm.value.focus()
  }, 100)
}
</script>

<style lang="scss" scoped>
.name-text-field {
  align-self: stretch;
  grid-template-columns: repeat(auto-fill, 1fr);
  grid-template-rows: 1fr auto;
}

.name-text-field:deep(.v-field--appended) {
  padding-inline-end: 0;
}

.name-text-field:deep(.v-field__input),
.name-text-field:deep(.v-field__append-inner) {
  padding-top: 0 !important;
}

.name-text-field:deep(.v-field__input) {
  min-height: auto;
  height: 100%;
  letter-spacing: 0;
}
</style>
