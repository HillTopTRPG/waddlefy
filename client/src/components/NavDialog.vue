<template>
  <v-dialog
    :model-value="modelValue"
    :contained="true"
    :fullscreen="true"
    :scrim="false"
    :no-click-animation="true"
    :persistent="true"
    class="my-16 nav-dialog"
    transition="slide-y-reverse-transition"
  >
    <v-card>
      <v-card-title class="d-flex justify-space-between align-center py-1">
        <span data-cy="title-text">{{ title }}</span>
        <v-btn icon="mdi-close" variant="text" size="small" style="font-size: 0.8em" @click="emits('close')"></v-btn>
      </v-card-title>
      <slot />
    </v-card>
  </v-dialog>
</template>

<script lang="ts" setup>
defineProps<{
  modelValue: boolean
  title: string
}>()

const emits = defineEmits<{
  (e: 'close'): void
}>()
</script>

<style lang="scss" scoped>
.v-dialog {
  margin-right: -1px;

  .v-card-title {
    user-select: none;
    background-color: rgba(var(--v-theme-secondary), 0.2);
    border-bottom: 1px solid rgba(100, 100, 100, 0.2);
  }
}
</style>
