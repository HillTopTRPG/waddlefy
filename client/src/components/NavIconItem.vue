<template>
  <v-list-item :style="toggle ? 'border-radius: 10px 0 0 10px;' : ''">
    <template v-slot:prepend>
      <v-icon :icon="`mdi-${icon}`" class="mr-6" />
    </template>
    <v-list-item-title v-if="!rail">{{ title }}</v-list-item-title>
    <v-list-item-subtitle v-if="!rail && subtitle">{{ subtitle }}</v-list-item-subtitle>
  </v-list-item>
</template>

<script setup lang="ts">
defineProps<{
  title: string
  subtitle?: string
  rail: boolean
  icon: string
  toggle: boolean
}>()
</script>
