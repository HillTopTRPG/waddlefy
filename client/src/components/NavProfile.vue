<script setup lang='ts'>
import { useRouter } from 'vue-router'
import { computed, inject, ref, watch } from 'vue'
import DeleteDialogButton from '~/pages/room/Dialog/DeleteDialogButton.vue'
import {User, UserTypeSelection, userTypeSelection} from "@/components/graphql/schema";
import {GraphQlKey, GraphQlStore} from "@/components/graphql/graphql";
const graphQlStore = inject<GraphQlStore>(GraphQlKey)

const props = defineProps<{
  room_uuid: string
  user_uuid?: string
  user_name?: string
  user_password?: string
  nav1?: string | 'room-info'
  nav2: string
  users: User[]
}>()

const emits = defineEmits<{
  (e: 'requireUserLogin'): void
  (e: 'close'): void
}>()

const user       = computed(() => props.users.find(u => u.id === props.user_uuid))
const targetUser = computed(() => props.users.find(u => u.id === props.nav1))

const router = useRouter()

const env = {
  router
}

const userName = ref('')
watch(user, () => {
  userName.value = user.value?.name || ''
}, {
        deep     : true,
        immediate: true,
      })

const userType = ref<UserTypeSelection | undefined>(userTypeSelection.find(s => s.value === user.value?.type))
const targetUserType = ref<UserTypeSelection | undefined>(userTypeSelection.find(s => s.value === targetUser.value?.type))
watch(() => user.value?.type, (value) => {
  userType.value = userTypeSelection.find(s => s.value === value) || userTypeSelection[1]
})

const updateUser = () => {
  const name      = userName.value
  const type = userType.value?.value
  if (!name || !props.user_uuid || !user.value) {
    return
  }
  return graphQlStore?.patchUser( { name, type })
}

const deleteUser = async () => {
  graphQlStore?.deleteUser()
}
</script>

<template>
  <v-overlay class='nav-contents' :contained='true' :model-value='nav2 === "profile"'>
    <v-list :nav='true' bg-color='transparent' v-if='user_uuid === nav1'>
      <v-list-subheader>あなたのアカウント情報</v-list-subheader>
      <v-list-item>
        <v-text-field
          v-model='userName'
          label='名前'
          hint='必須項目'
        />
      </v-list-item>
      <v-list-item>
        <v-select
          v-model='userType'
          :items='userTypeSelection'
          item-value='value'
          item-title='title'
          :hint='userType?.hint || ""'
          label='ユーザータイプ'
          persistent-hint
          return-object
        />
      </v-list-item>
      <v-list-item>
        <v-btn color='secondary' @click='updateUser()'>更新</v-btn>
        <delete-dialog-button button-text='削除' dialog-title='このユーザーを削除しますか？' @execute='deleteUser()' />
      </v-list-item>
    </v-list>
    <template v-else>
      {{ targetUser?.name }}({{ targetUserType?.title || 'プレイヤー' }})
    </template>
    <v-container class='d-flex align-center justify-center'>
      <v-btn icon='mdi-close' size='small' variant='tonal' @click='emits("close")'></v-btn>
    </v-container>
  </v-overlay>
</template>
