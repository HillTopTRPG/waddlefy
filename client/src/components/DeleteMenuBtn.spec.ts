import { createSimpleClassTest, factory, SimpleClassTestInfo } from '@/components/common'
import DeleteMenuBtn from '@/components/DeleteMenuBtn.vue'
import { VueWrapper } from '@vue/test-utils'

interface DeleteMenuBtnProps {
  type: string
  targetName: string
  classText?: string
  color?: string
}

declare module '@vue/runtime-core' {
  interface ComponentCustomProperties {
    opened: boolean
    deleteExecute: () => Promise<void>
    inputTargetName: string
  }
}

function factoryWrap(partialProps: Partial<DeleteMenuBtnProps>) {
  const props: DeleteMenuBtnProps = {
    type: '種別',
    targetName: '削除対象の名前',
    classText: 'additional-class',
    ...partialProps
  }
  return factory(DeleteMenuBtn, props)
}

describe('DeleteMenuBtn.vue', (): void => {
  describe('ボタンの確認', (): void => {
    describe('コンテンツの確認', (): void => {
      let wrapper: VueWrapper
      beforeEach(() => (wrapper = factoryWrap({})))
      afterEach(() => wrapper.unmount())

      it('スナップショットテスト', () => expect(wrapper.element).toMatchSnapshot())
      it('テキストの確認', () => expect(wrapper.text()).toEqual('label.delete-this'))
    })

    describe('classの確認', (): void => {
      describe('color無指定', (): void => {
        const patterns: SimpleClassTestInfo[] = [
          { title: 'colorがerrorであること', selector: '.v-btn', containClass: 'text-deleteText' },
          { title: 'decsityがdefaultであること', selector: '.v-btn', containClass: 'v-btn--density-default' },
          { title: 'variantがtextであること', selector: '.v-btn', containClass: 'v-btn--variant-text' },
          { title: 'sizeがdefaultであること', selector: '.v-btn', containClass: 'v-btn--size-default' },
          { title: 'iconがmdi-delete-outlineであること', selector: '.v-icon', containClass: 'mdi-delete-outline' },
          { title: '追加classが適用されていること', selector: '.v-btn', containClass: 'additional-class' },
          {
            title: 'テキストに下線がついていること',
            selector: '[data-cy=text]',
            containClass: 'text-decoration-underline'
          }
        ]
        patterns.forEach(createSimpleClassTest.bind(null, factoryWrap.bind(null, {})))
      })

      describe('colorをwarningに指定する', (): void => {
        const patterns: SimpleClassTestInfo[] = [
          { title: 'colorがwarningであること', selector: '.v-btn', containClass: 'text-warning' }
        ]
        patterns.forEach(createSimpleClassTest.bind(null, factoryWrap.bind(null, { color: 'warning' })))
      })
    })
  })

  describe('初期状態の確認', (): void => {
    let wrapper: VueWrapper
    beforeEach(() => (wrapper = factoryWrap({})))
    afterEach(() => wrapper.unmount())

    it('openedがfalseであること', () => expect(wrapper.vm.opened).toBe(false))
    it('inputTargetNameが空文字であること', () => expect(wrapper.vm.inputTargetName).toBe(''))
  })

  describe('deleteExecuteメソッドの確認', (): void => {
    let wrapper: VueWrapper
    beforeEach(() => (wrapper = factoryWrap({})))
    afterEach(() => wrapper.unmount())

    it('実行時にopenedがfalseになること', async () => {
      wrapper.vm.opened = true
      await wrapper.vm.deleteExecute()
      expect(wrapper.vm.opened).toBe(false)
    })

    it('実行時にexecuteイベントがemitされること', async () => {
      await wrapper.vm.deleteExecute()
      expect(wrapper.emitted()).toHaveProperty('execute')
    })

    it('実行時にinputTargetNameが空になること', async () => {
      wrapper.vm.inputTargetName = 'test'
      await wrapper.vm.deleteExecute()
      expect(wrapper.vm.inputTargetName).toBe('')
    })
  })
})
