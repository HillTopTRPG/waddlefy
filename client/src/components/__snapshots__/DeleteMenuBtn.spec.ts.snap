// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DeleteMenuBtn.vue > ボタンの確認 > コンテンツの確認 > スナップショットテスト 1`] = `
<div
  data-v-app=""
>
  
  
  <button
    aria-expanded="false"
    aria-haspopup="menu"
    aria-owns="v-menu-0"
    class="v-btn v-theme--light text-error v-btn--density-default v-btn--size-default v-btn--variant-text additional-class"
    data-cy="open-btn"
    type="button"
  >
    
    <span
      class="v-btn__overlay"
    />
    <span
      class="v-btn__underlay"
    />
    
    <!---->
    <span
      class="v-btn__content"
      data-no-activator=""
    >
      
      
      <i
        aria-hidden="true"
        class="mdi-delete-outline mdi v-icon notranslate v-theme--light v-icon--size-default"
      />
      <span
        class="text-decoration-underline"
        data-cy="text"
      >
        この種別を削除
      </span>
      
      
    </span>
    <!---->
    <!---->
  </button>
  
  <!---->
  
</div>
`;
